flink-connectors/flink-connector-cassandra/src/main/java/org/apache/flink/streaming/connectors/cassandra/CassandraSinkBase.java

```diff
@@ -37,29 +41,49 @@
  *
  * @param <IN> Type of the elements emitted by this sink
  */
-public abstract class CassandraSinkBase<IN, V> extends RichSinkFunction<IN> {
+public abstract class CassandraSinkBase<IN, V> extends RichSinkFunction<IN> implements CheckpointedFunction {
 	protected final Logger log = LoggerFactory.getLogger(getClass());
 	protected transient Cluster cluster;
 	protected transient Session session;
 
-	protected transient volatile Throwable exception;
+	protected transient volatile Throwable asyncError;
 	protected transient FutureCallback<V> callback;
 
-	private final ClusterBuilder builder;
+	protected final ClusterBuilder builder;
 
 	private final AtomicInteger updatesPending = new AtomicInteger();
 
+	/**
+	 * If true, the producer will wait until all outstanding action requests have been sent to C*.
+	 */
+	private boolean flushOnCheckpoint = true;
```

Comment:
this should be a constructor argument and final.

Reply:
Should we default this setting as true then? Making it final does not allow to reset the flag after obj is constructed. Further, should this behavior be applied to all other SinkBase / Producer classes in the future?

Reply:
The default should be true. Every CassandraSinkBase subclass should accept a constructor argument that is passed on to the CassandraSinkBase constructor. Additionally, the CassandraSinkBuilder should be extended to allow setting this flag to false if desired.

Reply:
Got your points on the behaviors to subclasses to CassandraSinkBase, but I was wondering if some of these 'common behavior' should be or would be unified across different data sink, KafKa ES, etc.  They all currently have slightly different implantation though.

Reply:
Ideally all sinks should behave the same way, but that would be out of scope for PR in particular.